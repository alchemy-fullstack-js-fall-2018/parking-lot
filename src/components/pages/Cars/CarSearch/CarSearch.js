import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import CarList from '../CarList';
import { ROUTES } from '../../../../routes';
import queryString from 'querystring';

export default class CarSearch extends PureComponent {

  static propTypes = {
    cars: PropTypes.array.isRequired,
    updateSearch: PropTypes.func.isRequired
  };

  componentDidMount() {
    this.doSearch();
  }

  componentDidUpdate(prevProps, prevState) {
    if(prevProps.location.search !== this.props.location.search) {
      this.doSearch();
    }
  }

  getQuery = () => {
    return queryString.parse(this.props.location.search.slice(1));
  }

  doSearch = () => {
    const { q } = this.getQuery();
    this.props.updateSearch(q);
  };

  updateHistory = (options) => {
    const q = options.q || '';
    const searchQuery = queryString.stringify({ q });
    this.props.history.push({
      pathName: ROUTES.CAR_SEARCH_ALL.path,
      search: searchQuery
    });
  };

  updateSearch = event => {
    event.preventDefault();
    const q = document.getElementById('q').value;
    this.doSearch;
    this.updateHistory({ q });
  };

  render() {

    const { cars } = this.props;

    return (
      <div>
        <form onSubmit={this.updateSearch}>
          <label htmlFor="q">Search</label>
          <input id="q" name="q" type="text" />
          <button type="submit">Search</button>
        </form>
        <CarList cars={cars}/>
      </div>
    );
  }

}
