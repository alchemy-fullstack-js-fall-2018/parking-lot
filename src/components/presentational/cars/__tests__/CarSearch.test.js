import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import CarSearch from '../CarSearch.jsx';

jest.mock('../../../../routes');

// jest.mock('../../../../routes', () => ({
//   ROUTES: {
//     CAR_DETAIL: {
//       linkTo: () => '/'
//     },
//     CAR_SEARCH_ALL: {
//       linkTo: () => '/asdf'
//     }
//   }
// }));

describe('<CarSearch />', () => {
  test('renders', () => {
    const updateSearchFn = jest.fn();
    const cars = [
      {
        plate: 'abcdef',
        make: 'Ford',
        model: 'F150',
        inLot: false,
        lotUses: 5,
        id: '123-123-123-123',
      },
      {
        plate: 'xyzabc',
        make: 'Chevy',
        model: 'S10',
        inLot: true,
        lotUses: 8,
        id: '321-321-321-321',
      }
    ];
    const wrapper = shallow(
      <CarSearch cars={cars} updateSearch={updateSearchFn} history={[]} location={{ search:'asdf' }} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
