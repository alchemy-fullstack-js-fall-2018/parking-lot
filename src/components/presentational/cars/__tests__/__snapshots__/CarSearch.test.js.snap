// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarSearch /> renders 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="q"
    >
      Search
    </label>
    <input
      id="q"
      name="q"
      type="text"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
  <CarList
    cars={
      Array [
        Object {
          "id": "123-123-123-123",
          "inLot": false,
          "lotUses": 5,
          "make": "Ford",
          "model": "F150",
          "plate": "abcdef",
        },
        Object {
          "id": "321-321-321-321",
          "inLot": true,
          "lotUses": 8,
          "make": "Chevy",
          "model": "S10",
          "plate": "xyzabc",
        },
      ]
    }
  />
</div>
`;
